src/routes/reservation/index.ts

```diff
@@ -0,0 +1,252 @@
+/* eslint-disable no-unused-vars */
+
+import * as awsx from '@pulumi/awsx';
+import { v4 as uuidv4 } from 'uuid';
+import { DynamoDB } from 'aws-sdk';
+import { marshall, parseBody, buildApiResponse, add500Handler, getUserId } from '$src/apiGatewayUtilities';
+import { query } from '$src/dynamodb/utils';
+import { createDynamo } from '$src/initAWS';
+
+function toResponse(entry: DynamoDB.AttributeMap) {
+  return {
+    id: entry.id.S,
+    propertyId: entry.propertyId.S,
+    customerId: entry.customerId.S,
+    bookedRoomsNumber: entry.bookedRoomsNumber.S,
+    beginDate: entry.beginDate.S,
+    endDate: entry.endDate.S,
+  };
+}
+
+function toArrayResponse(items: DynamoDB.Types.ItemList) {
+  return items.map((item) => toResponse(item));
+}
+
+export function getCustomerReservations() {
+  const dynamo = createDynamo();
+
+  const handler = async (event: awsx.apigateway.Request) => {
+    const customerId = getUserId(event);
+
+    const params = {
+      TableName: 'reservation',
+      KeyConditionExpression: 'customerId = :customerId',
+      ExpressionAttributeValues: {
+        ':customerId': { S: customerId },
+      },
+      Limit: 1000,
+    };
+
+    const reservations = await query(dynamo, params);
+
+    return buildApiResponse(200, toArrayResponse(reservations));
+  };
+
+  return add500Handler(handler);
+}
+
+function splitRange(beginDate: Date, endDate: Date) {
+  const dates = [];
+  const currDate = new Date(beginDate);
+
+  while (currDate < endDate){
+    dates.push(new Date(currDate));
+    currDate.setDate(currDate.getDate() + 1);
+  };
+
+  dates.push(new Date(endDate));
+
+  return dates;
+}
+
+function createGetLockedRoomsCountHandler(beginDate: Date, endDate: Date) {
+  return (items: DynamoDB.Types.ItemList) => {
```

Comment:
if we change this to expect an object with startDate and endDate, we can unit-test this funciton properly with different combinations of dates